### PR TITLE
[INLONG-11546][SDK] Support async and sync report dirty data

### LINKED ISSUE #11546: [Feature][SDK] Support async and sync report dirty data
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
Fixes #11546 

### Motivation

Provide async and sync way to report dirty data

### Modifications

Add sendDirtyMessage and sendDirtyMessageAsync method

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by luchunliang — APPROVED** [review]

**Review by luchunliang — DISMISSED** [review]

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/InlongSdkDirtySender.java [hunk 1]
```diff
@@ -74,8 +73,14 @@ public void init() throws Exception {
         log.info("init InlongSdkDirtySink successfully, target group={}, stream={}", inlongGroupId, inlongStreamId);
     }
 
-    public void sendDirtyMessage(DirtyMessageWrapper messageWrapper) throws InterruptedException {
-        dirtyDataQueue.offer(messageWrapper, 10, TimeUnit.SECONDS);
+    public void sendDirtyMessageSync(DirtyMessageWrapper messageWrapper) throws InterruptedException {
```
Comment (aloyszhang) [inline inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/InlongSdkDirtySender.java:76]:
```suggestion
    public void sendDirtyMessage(DirtyMessageWrapper messageWrapper) throws InterruptedException {
```

Comment (vernedeng) [reply inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/InlongSdkDirtySender.java:76]: fixed, thx

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/InlongSdkDirtySender.java [hunk 2]
```diff
@@ -75,7 +74,15 @@ public void init() throws Exception {
...
     public void sendDirtyMessage(DirtyMessageWrapper messageWrapper) throws InterruptedException {
-        dirtyDataQueue.offer(messageWrapper, 10, TimeUnit.SECONDS);
+        dirtyDataQueue.put(messageWrapper);
+    }
+
+    public boolean sendDirtyMessageAsync(DirtyMessageWrapper messageWrapper) {
+        boolean result = dirtyDataQueue.offer(messageWrapper);
+        if (!result) {
+            log.warn("send dirty message async queue is full, you can increase maxCallbackSize");
+        }
```
[the remark this answers is not in the document]
Comment (vernedeng) [reply inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/InlongSdkDirtySender.java]: just remove this log

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/InlongSdkDirtySender.java [hunk 3]
```diff
@@ -75,7 +74,15 @@ public void init() throws Exception {
     }
 
     public void sendDirtyMessage(DirtyMessageWrapper messageWrapper) throws InterruptedException {
-        dirtyDataQueue.offer(messageWrapper, 10, TimeUnit.SECONDS);
+        dirtyDataQueue.put(messageWrapper);
+    }
```
[the remark this answers is not in the document]
Comment (vernedeng) [reply inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/InlongSdkDirtySender.java:78]: we provide an async method sendDirtyMessageAsync

**Review by baomingyu — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**